application/src/Controller/Site/IndexController.php

```diff
@@ -37,29 +37,57 @@ public function indexAction()
     public function searchAction()
     {
         $fulltextQuery = $this->params()->fromQuery('fulltext_search');
+
+        $siteSettings = $this->siteSettings();
+
+        $resourceNames = $siteSettings->get('search_resource_names', ['site_pages', 'items', 'item_sets', 'media']);
```

Comment:
I think we'd probably prefer if this was defaulted to site_pages and items only (and the same with the default value in the form), so that it's not a behavior change over the current page unless you explicitly go change the setting.